src/wasm/wasm-type.cpp

```diff
@@ -32,60 +32,57 @@ template<> class hash<vector<wasm::Type>> {
   size_t operator()(const vector<wasm::Type>& types) const {
     auto res = hash<size_t>{}(types.size());
     for (auto t : types) {
-      wasm::hash_combine(res, t.getID());
+      res = wasm::rehash(res, t.getID());
```

Comment:
This is the old code and the only occurrence where `t.getID()` is not piped through `std::hash<uint64_t>{}(...)`. The other occasions are odd in that piping through `std::hash` gives us a `size_t`, and I do not understand how this is supposed to work on 32-bit platforms.

In this function particularly, we are first hashing a `size_t` to obtain a `size_t`, then rehashing that to `uint64_t` before ultimately returning a `size_t`.